### PR TITLE
chore(deps): cmake/ninja: add new versions

### PR DESCRIPTION
**Pull request by novas0x2a**

### CODE REVIEW

File: toolchains/prebuilt_toolchains.py [hunk 1]
```diff
@@ -166,6 +172,7 @@
 }
 
 NINJA_VERSIONS = (
+    "1.12.1",
```
Comment (jsharpe) [inline toolchains/prebuilt_toolchains.py:175]: Is the update to the ninja repo list missing for this update?

**Comment by jsharpe** [comment]
Apologies I broke main by accidently committing directly to the main branch from github.dev - revert is in #1213

File: toolchains/prebuilt_toolchains.py (hunk 1 above)
Comment (novas0x2a) [reply toolchains/prebuilt_toolchains.py:175]: oh, I guess it is; when I run the script by hand and try to commit the result, the repo hooks don't let me; when I let the repo hooks run the script themselves; it looks like they don't stage the ninja file, just the cmake file

Comment (novas0x2a) [reply toolchains/prebuilt_toolchains.py:175]: Oh, I see what's happening. The hook requires using buildifier 6.1.0, but my buildifier is 7.1.2, and they produce different output. PR updated.

**Comment by novas0x2a** [comment]
> Apologies I broke main by accidently committing directly to the main branch from github.dev - revert is in #1213

No worries, rebased :)

**Review by jsharpe — APPROVED** [review]

**Pull request merged**